### ISSUE TITLE
Audit policy check incorrectly flagging missing policies

### ISSUE THREAD

**Issue by ralish** [vletoux/pingcastle#30]
The new audit policy configuration check appears to be a little buggy. I'm frequently seeing numerous audit policies being flagged as missing in the *Check if there is the expected audit policy on domain controllers* check. However, they're definitely applied (verified by `auditpol /get /category:*`) and PingCastle does successfully discover all the audit policies defined in GPOs as they're listed in the *Audit Settings* summary later in the report.

I suspect PingCastle might be failing to factor in merging of advanced audit policies across multiple GPOs? All testing performed with PingCastle v2.8.1.0. And as always, keep up the great work :-)

**Comment by vletoux** [comment]
The program collects information about GPO.
It then filter if they are applied to DC.
Then it checks in the list of required policies for missing items.
The algorithm is here: https://github.com/vletoux/pingcastle/blob/master/Healthcheck/Rules/HeatlcheckRuleAnomalyAuditDC.cs

So either GPO is not accessible by the program or there is a bug.
But I need to reproduce the problem in order to understand it, then fix it.
In short, if you notify a GPO that is not seen correctly, please zip the GPO files and send them to support@PingCastle.com

**Comment by ralish** [comment]
@vletoux Taking a quick look at the code it appears the algorithm is only factoring in GPOs directly applied to [the *Domain Controllers* OU](https://github.com/vletoux/pingcastle/blob/master/Healthcheck/Rules/HeatlcheckRuleAnomalyAuditDC.cs#L158) or a child OU? That's not going to work for GPOs linked to the top-level `domain` object, assuming inheritance is not blocked to the *Domain Controllers* OU.

**Comment by vletoux** [comment]
You’re right.
That’s why everybody should test beta versions instead waiting for 1 month after the release ...

**Comment by ralish** [comment]
Confirming this appears to be fixed in v2.9.0.0 Beta. 💯
